cmd/swagger/commands/serve_ui.go

```diff
@@ -0,0 +1,57 @@
+package commands
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/go-swagger/go-swagger/spec"
+	"github.com/rs/cors"
+	"github.com/toqueteos/webbrowser"
+)
+
+//ServeUI Show given swagger spec in swagger-ui
+type ServeUI struct {
+}
+
+// Execute this command
+func (s *ServeUI) Execute(args []string) error {
+
+	if len(args) == 0 {
+		return errors.New("The serve-ui command requires the swagger document url to be specified")
+	}
+	swaggerDoc := args[0]
+	specDoc, err := spec.Load(swaggerDoc)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	serveUI(specDoc)
+	return nil
+}
+
+func serveUI(doc *spec.Document) error {
+
+	// parse the url and open in default browser
+	u, err := url.Parse("http://petstore.swagger.io/")
+	if err != nil {
+		return err
+	}
+	q := u.Query()
+	q.Add("url", "http://localhost:8080/swagger.json")
+	u.RawQuery = q.Encode()
+	webbrowser.Open(u.String())
+
+	// serve the swagger.json
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(doc.Raw()))
+	})
+
+	handler := cors.Default().Handler(mux)
+
+	http.ListenAndServe(":8080", handler)
```

Comment:
can you make this a random port so that there aren't any collisions?
Also print out the full url to petstore.swagger.io?url=specurl 
That allows for people to cmd+click or ctrl click the url to go there even fit he browser didn't open it. Might also be good to make the auto browser opening a flag as well as a flag for the base url for swagger ui